core/vm/interpreter.go

```diff
@@ -91,46 +89,45 @@ type EVMInterpreter struct {
 
 	readOnly   bool   // Whether to throw on stateful modifications
 	returnData []byte // Last CALL's return data for subsequent reuse
+
+	jt *JumpTable // EVM instruction table
 }
 
 // NewEVMInterpreter returns a new instance of the Interpreter.
 func NewEVMInterpreter(evm *EVM, cfg Config) *EVMInterpreter {
-	// We use the STOP instruction whether to see
-	// the jump table was initialised. If it was not
-	// we'll set the default jump table.
-	if !cfg.JumpTable[STOP].valid {
-		var jt JumpTable
-		switch {
-		case evm.chainRules.IsYoloV1:
-			jt = yoloV1InstructionSet
-		case evm.chainRules.IsIstanbul:
-			jt = istanbulInstructionSet
-		case evm.chainRules.IsConstantinople:
-			jt = constantinopleInstructionSet
-		case evm.chainRules.IsByzantium:
-			jt = byzantiumInstructionSet
-		case evm.chainRules.IsEIP158:
-			jt = spuriousDragonInstructionSet
-		case evm.chainRules.IsEIP150:
-			jt = tangerineWhistleInstructionSet
-		case evm.chainRules.IsHomestead:
-			jt = homesteadInstructionSet
-		default:
-			jt = frontierInstructionSet
-		}
-		for i, eip := range cfg.ExtraEips {
-			if err := EnableEIP(eip, &jt); err != nil {
-				// Disable it, so caller can check if it's activated or not
-				cfg.ExtraEips = append(cfg.ExtraEips[:i], cfg.ExtraEips[i+1:]...)
-				log.Error("EIP activation failed", "eip", eip, "error", err)
-			}
+	var jt *JumpTable
```

Comment:
There is a nicer way to do this: if you change the type of `jt` to just `JumpTable`, you don't need to have all those `&`s below and you remove the extra copy in the loop below.

Reply:
Not sure if that defeats the optimization though :)

Reply:
I think that defeats the entire purpose of the PR... ?

Reply:
Hmm, but you do have to copy it when there are ExtraEips. 

I'm really wondering about the performance impact of this change.
I think the idea is to speed up NewEVMInterpreter, but AFAIK that's not
really the bottleneck for anything.

Reply:
Anyway, you can disregard my comment. We should still measure what the impact of this is, can't be that much.